src/engines/interactions-engine.js

```diff
@@ -193,7 +193,7 @@ export class InteractionsEngine extends EventEmitter {
             if (hoveredInstance) {
                 const offsetTop = hoveredInstance.renderEngine.position;
 
-                return hoveredInstance.hitRegions.find(({ x, y, w, h }) => (
+                return hoveredInstance.hitRegions.slice().reverse().find(({ x, y, w, h }) => (
```

Comment:
It is too unoptimized.(

Reply:
I agree, copying this all the time can cause perf issues
I still remember times when I started using the first version of the chart and tried to visualize 1milion of function calls.
You put a lot of effort to make it fast at that scale

Reply:
the generation comparisons with the demo didn't seem any slower.  Still remains that overlapping spans with the separate queues clobber each other visually and interactively.  These changes DO fix this issue.  I don't need to visualize millions of spans however, only thousands...

Reply:
yah.. I can see this, will need to rewrite to just find from the end.  this seems a quick/dirty solution on review.

Reply:
fixed to use a findLast function